File: strings/reverse_words_in_a_string_iii.rb
# @param {String} s
# @return {String}
def reverse_words(s)
    s.split.map(&:reverse).join(" ")
end

# 557. Reverse Words in a String III
# https://leetcode.com/problems/reverse-words-in-a-string-iii/

# Complexity Analysis
# Time complexity: O(n), Just 1 loop of length of string
# Space complexity: O(1)

File: strings/reverse_words_in_a_string_iii.py
class Solution:
    def reverseWords(self, s: str) -> str:
        return ' '.join(list(map(lambda x : x[::-1], s.split())))


# https://leetcode.com/problems/reverse-words-in-a-string-ii/
# 186. Reverse Words in a String II

# Time: O(n)
# Space: O(n)
